### ISSUE TITLE
Make a button next to entity class to create a new entity (no need to open right click menu)

### ISSUE THREAD

**Issue by nelliputkonen** [spine-tools/Spine-Toolbox#2788]

**Comment by PiispaH** [comment]
> Add new entities by writing (or copy-pasting) a new entity name in the last row of Parameter value widget.

Currently there is this "autofill" functionality that if there are existing entities that start with the text that is currently written in the entity_byname box, pressing enter, tab or clicking away will select the first item in the suggestion list.

![image](https://github.com/spine-tools/Spine-Toolbox/assets/98282892/4627a070-8c1a-4677-adb6-4128913a16b9)

If I have for example entity "AA" and I'm trying to create a new entity "A" in the same entity class by typing, the autofill won't allow this. I feel like the autofill is a very nice thing to have. This could be done in a way that if the autofill doesn't come up with anything, only then a new entity is created. This is only a small edge case that I'm describing anyhow and in those cases the user would probably realize to add the entity through the Entity Tree.

When data is pasted to the table this probably won't be a problem.

Thoughts @soininen @jkiviluo @nelliputkonen ?

**Comment by nelliputkonen** [comment]
> > 'Entity groups' could simply be text field like 'description'
> 
> So the user could write some name for a Entity group here and if the group doesn't exist yet, it will be created, and the entity gets automatically added there? If the group already exists the entity will just be added there? Or are these entity groups something different than the ones that exist currently? @nelliputkonen

Yes, this would my idea. However I'm open for better suggestions. The idea is just mainly to make an easy way to organize your entities besides alphabetical order.

**Comment by soininen** [comment]
The "entity group" concept does not sound like the entity groups we currently have. Sounds more like "entity category" or such, am I correct?

**Comment by PiispaH** [comment]
Yeah, since Entity groups can only contain entities from one class, it would probably be more useful to have more general "groups" that can have entities from different classes. That way the filtering and ordering would be more powerful.

**Comment by jkiviluo** [comment]
I don't think we at this point want to create a new thing in addition to existing entity groups - my thought was that this would make it easier for people to start using the existing entity group thing for something useful like categorizing stuff. At the moment SpineOpt uses groups for some things, but it's not widespread (mostly because it's limited to be within entity_class).

The idea of having cross-entity-class groups (they could be called 'bags' like in our data structure plan) is enticing, but I don't think we have the resources for that in near future. It would cause lot of changes across the code-base.

One thing to note is that one entity can be in multiple entity_groups. I think it just means that one entity might show up multiple times when the entities are ordered by entity_group. When entering new entities, there is no need to have the option to include it in multiple groups even that is possible in the entity group dialog.

But that said, if there are good arguments of having entity category label instead of entity_group here, would be good to understand. I'm just hoping for a low effort improvement (just adding an easier way to assign entity to group while creating it) - and then being able to use it in filtering/ordering.

**Comment by jkiviluo** [comment]
Found this one, that is quite related: https://github.com/spine-tools/Spine-Toolbox/issues/2370. No immediate need to change anything, but this second issue could be a follow-up relatively soon.

**Comment by soininen** [comment]
> I don't think we at this point want to create a new thing in addition to existing entity groups...

Fair enough, thanks for clarifying! I think I am on the same boat now.

**Comment by PiispaH** [comment]
Everything should be implemented now.
